### PR TITLE
Add Heterogeneous Design Proposal 

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

ref https://github.com/pingcap/tidb-operator/issues/2240

Add Design document for Heterogeneous Design.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  cluster:
+    name: <your-tidbcluster-name>
+    namespace: <your-tidbcluster-namepace>
+  <some tikvspec.....>
+```
+
+
+# TODO
+
+For more detail, please [see also](https://docs.google.com/document/d/1MV2bcsCjyYvfCCtwyc8-E18Z69qP_3-FKztpGj4ByEg/edit?usp=sharing)
```
[the remark this answers is not in the document]
Comment (cofyc) [reply docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:61]: hope we can migrate all details here

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Heterogeneous design for TidbCluster
+
+This document presents a design for the Heterogeneous components in one kind for `TidbCluster`.
+
+## Motivation
+
+Currently, the spec for the `TidbCluster` describe a groups of `PD`/`TiKV`/`TiDB` instances. For each kind of components, 
+their spec are all the same. This design is easy to use and also cover the most cases for using tidb cluster.
+
+However, as a distributed databases with clear and multi layers, the components in each layer could be different from
```
Comment (cofyc) [inline docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:10]:
```suggestion
However, as a distributed database with clear and multilayers, the components in each layer could be different from 
```

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Motivation
+
+Currently, the spec for the `TidbCluster` describe a groups of `PD`/`TiKV`/`TiDB` instances. For each kind of components, 
+their spec are all the same. This design is easy to use and also cover the most cases for using tidb cluster.
+
+However, as a distributed databases with clear and multi layers, the components in each layer could be different from 
+each other to meet the different requirements.
+
+For a example, as sql layer, tidb components could be composed of multiple instances with different resource requests
```
Comment (cofyc) [inline docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:13]:
```suggestion
For example, as the SQL layer, tidb components could be composed of multiple instances with different resource requests 
```

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+## Motivation
+
+Currently, the spec for the `TidbCluster` describe a groups of `PD`/`TiKV`/`TiDB` instances. For each kind of components, 
+their spec are all the same. This design is easy to use and also cover the most cases for using tidb cluster.
+
+However, as a distributed databases with clear and multi layers, the components in each layer could be different from 
+each other to meet the different requirements.
+
+For a example, as sql layer, tidb components could be composed of multiple instances with different resource requests 
+and configuration to handle the different workloads (AP/TP query). For the storage layer, tikv components could be
```
Comment (cofyc) [inline docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:14]:
```suggestion
and configurations to handle the different workloads (AP/TP query). For the storage layer, tikv components could be 
```

File: docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+For `TiKVGroup`:
+
+```golang
+type TiKVGroup struct {
+	Spec   TiKVGroupSpec
+	Status TiKVGroupStatus
+}
+
+type TiKVGroupSpec struct {
+    TiKVGrupSpec
```
Comment (DanielZhangQD) [inline docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md:31]: Is the `Grup` typo?

Comment (Yisaer) [reply docs/design-proposals/2020-06-12-heterogeneous-design-for-tidb-cluister.md]: updated.

**Pull request merged**